File: BubbleSort.js
function bubbleSort(array) {
    const n = array.length;
    for(let i = 0; i < n - 1; i++) {
        for(let j = 0; j < n - i - 1; j++) {
            if(array[j] > array[j + 1]) {
                let swap = array[j];
                array[j] = array[j + 1];
                array[j + 1] = swap;
            }
        }
    }
    console.log("The array of integers after the Bubble Sort is: ");
    console.log(array.join(" "));
}

function main() {
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    rl.question("Enter the number of elements to be entered in the array: ", function(n) {
        n = parseInt(n);
        let array = [];
        console.log(`Enter ${n} elements:`);
        rl.on('line', function(input) {
            array.push(parseInt(input));
            if(array.length === n) {
                console.log("The array of integers before BubbleSort:");
                console.log(array.join(" "));
                bubbleSort(array);
                rl.close();
            }
        });
    });
}

main();

File: BubbleSort.py
def bubble_sort(array):
    n = len(array)
    for i in range(n - 1):
        for j in range(n - i - 1):
            if array[j] > array[j + 1]:
                array[j], array[j + 1] = array[j + 1], array[j]
    print("The array of integers after the Bubble Sort is:")
    print(*array)

def main():
    n = int(input("Enter the number of elements to be entered in the array: "))
    array = []
    print(f"Enter {n} elements:")
    for i in range(n):
        element = int(input(f"Enter element {i + 1}: "))
        array.append(element)
    print("The array of integers before BubbleSort:")
    print(*array)
    bubble_sort(array)

if __name__ == "__main__":
    main()
